File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, A: str) -> str:
        stack = deque([]);

        for char in A:

            if len(stack) == 0 :
                stack.append(char);
            elif stack[-1] != char:

                stack.append(char);
            else:
                stack.pop();
        return "".join(stack);

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        
        StringBuilder sb = new StringBuilder();

        for (char ch : s.toCharArray()){
            
            int n = sb.length();

            if (n == 0 || sb.charAt(n - 1) != ch){
                sb.append(ch);
            }
            else {
                sb.deleteCharAt(n - 1);
            } 
        }
        return sb.toString();
    }
}